### PR TITLE
[v7r0] Add documentation for "overrides" folder in WebApp pages

### PR DESCRIPTION
**Pull request by chrisburr**
See https://github.com/DIRACGrid/management/pull/13.

I also tidied up the formatting the WebApp docs.

BEGINRELEASENOTES

ENDRELEASENOTES

### CODE REVIEW

**Comment by andresailer** [comment]
Wrong base branch, integration instead of rel-v7r0?

File: docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst [hunk 1]
```diff
@@ -27,119 +27,148 @@ Each application consists of two parts:
...
+  └── WebApp
+      ├── __init__.py
+      ├── handler
+      │   # Server side implementations of the framework and all the applications
+      │   └── __init__.py
+      ├── static
+      │   # Content that can be loaded by the client side such as JavaScript files, images and css files
+      │   └── <Module name such as DIRAC, LHCbDIRAC, WebAppDIRAC>
+      │       # Client side implementation of each application
+      │       ├── Application
```
Comment (andresailer) [inline docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst:42]: Application[12] are subfolders of DIRAC/LHCbDIRAC, no? so there should be another of the `|_` ?

File: docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst [hunk 2]
```diff
@@ -228,251 +259,261 @@ by the server is shown in the textarea.
...
+
+  ::
+
+    WebApp
+    ├── handler
+    |   └── MyAppHandler.py
+    └── static
+        └── DIRAC
+            └── MyApp
+                ├── build
```
Comment (andresailer) [inline docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst:275]: add `overrides` here?

File: docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst [hunk 3]
```diff
@@ -482,32 +523,34 @@ In Chrome you can use developer tools.
...
 
-      
-Before you can use the compiled version of the JavaScript files, you have to compiled them first. 
-For this reason you have to execute the python script **dirac-webapp-compile**. 
-In order to run the script, you have to download and install a tool called Sencha Cmd ( `<http://www.sencha.com/products/sencha-cmd/download>`_ ). 
-You can also refer to `<http://docs.sencha.com/extjs/4.2.1/#!/guide/command>`_ and read 
+Before you can use the compiled version of the JavaScript files, you have to compiled them first.
+For this reason you have to execute the python script **dirac-webapp-compile**.
+In order to run the script, you have to download and install a tool called Sencha Cmd ( `<https://www.sencha.com/products/sencha-cmd/download>`_ ).
+You can also refer to `<https://docs.sencha.com/extjs/4.2.1/#!/guide/command>`_ and read
```
Comment (andresailer) [inline docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst:553]: Is this version still the right one?

File: docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst [hunk 4]
```diff
@@ -535,45 +580,47 @@ Next, when defining the buildUI method, first of all the parent buildUI has to b
...
 
-We already prepared a simple example using predefined widgets
-(You can found more information `<https://hostname/DIRAC/static/doc/index.html>`_ and 
-you can have a look the code in github: (`<https://github.com/DIRACGrid/WebAppDIRAC/tree/integration/WebApp/static/DIRAC>`_).
+We already prepared a simple example using predefined widgets named ``ExampleApp`` that can be found `on github <https://github.com/DIRACGrid/WebAppDIRAC/tree/integration/WebApp/static/DIRAC/ExampleApp>`_.
+
+NOTE: Please make sure that your application will compile. For this you should run:
 
-NOTE: Please make sure that your application will compile. You have to use::
+.. code-block:: python
```
Comment (andresailer) [inline docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst:624]:
```suggestion
.. code-block:: bash
```

File: docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst [hunk 5]
```diff
@@ -228,251 +259,261 @@ by the server is shown in the textarea.
...
+
+5. Throughout all the code, especially in the method buildUI, there are several components created in order to structure the user interface. Therefore, you have to append all the classes used within the **DIRAC.MyApp.classes.MyApp** requires definition. In our case the list of requires would look like:
+
+  .. code-block:: javascript
+
+    requires:   ['Ext.panel.Panel', 'Ext.form.field.TextArea', 'Ext.Button', 'Ext.toolbar.Toolbar', 'Ext.data.JsonStore', 'Ext.grid.Panel']
+
+
+6. In order to have the application within the list of applications, you have to open the **web.cfg** file
+  located into the root. There you have to add new registration line within the **Schema/Applications** section:
```
Comment (andresailer) [inline docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst:492]: I guess this needs one more space of indentation?

File: docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst [hunk 6]
```diff
@@ -228,251 +259,261 @@ by the server is shown in the textarea.
...
-    
-    
-  6. In order to have the application within the list of applications, you have to open the **web.cfg** file 
-  located into the root. There you have to add new registration line within the **Schema/Applications** section::
-      
-      WebApp
+
+
+2. Now we have to create the folder structure for the CS. The main folder of the **MyApp** application have
+  to be located in a namespace folder. Let name that namespace folder DIRAC and place it in the **[root]/static/** folder.
```
Comment (andresailer) [inline docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst:265]: One more space of indentation?

File: docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst [hunk 7]
```diff
@@ -27,119 +27,148 @@ Each application consists of two parts:
...
-         * __init__.py
-      * **static**: contains all the static content that can be loaded by the client side such as JavaScript files, images and css files
-         * <Module name folder such as DIRAC, LHCbDIRAC, WebAppDIRAC>: contains the client side implementation of each application
-            * Application 1
-            * Application 2
-      * **template**: contains all the templates used by the files in the handler folder
+::
+
+  <Module name such as DIRAC, LHCbDIRAC, WebAppDIRAC>
+  └── WebApp
```
Comment (fstagni) [inline docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst:33]: Are all these characters the so-called box-drawing-characters? Isn't there a more straight-forward way of achieving the same result?

Comment (chrisburr) [reply docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst:33]: I've just copied that characters used by `tree`. I think it's clearer than the [current bulleted-list style](https://dirac.readthedocs.io/en/latest/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.html) but I'm open to other suggestions.

Comment (fstagni) [reply docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst:33]: It's definitely cleaner but just I don't have these characters on the keyboard so the fastest for editing is a copy-paste. I was just wondering if this is the RST-suggested way of working.

Comment (chrisburr) [reply docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst:33]: I didn't mange to find anything better and the Sphinx documentation doesn't include anything like this to use as a reference. The most common other way I've seen is to not bother with the extra characters and use block formatting+indentation. For example from `pytest`:
```
setup.py
mypkg/
    __init__.py
    app.py
    view.py
tests/
    test_app.py
    test_view.py
    ...
```
I tried this but it was harder to keep the distinction between the inline comments and file/folder names.

Comment (andresailer) [reply docs/source/DeveloperGuide/WebAppDIRAC/DevelopWebApp/index.rst:33]: There is apparently also the "ascii" flag for tree using 
https://meta.stackexchange.com/a/147474

**Pull request merged**